nixos/modules/installer/tools/nixos-install.sh

```diff
@@ -132,8 +132,9 @@ if [[ -z $noBootLoader ]]; then
     echo "installing the boot loader..."
     # Grub needs an mtab.
     ln -sfn /proc/mounts $mountPoint/etc/mtab
-    NIXOS_INSTALL_BOOTLOADER=1 nixos-enter --root "$mountPoint" -- /run/current-system/bin/switch-to-configuration boot
+    export NIXOS_INSTALL_BOOTLOADER=1
 fi
+nixos-enter --root "$mountPoint" -- /run/current-system/bin/switch-to-configuration boot
```

Comment:
Why was this moved out of the `if` ? Why do we need to run `switch-to-configuration boot` ?

Reply:
We have to ensure that _/sbin/init_ is available, otherwise _machinectl_ do not know per default what to start.
The NixOS option _boot.loader.initScript.enable_ setups _system.build.installBootLoader_, which creates /sbin/init.
I called _switch-to-configuration_ to get the script executed.

Do you see a better way to solve this issue?

Reply:
Hm, why not simply *not* pass the `--no-bootloader` flag? Seems to me that you *do* want to install the boot loader (i.e. the creation of `/sbin/init`) here.

Reply:
because a container machinectl _directory_ will not have a `/boot` partition, but it will need an `/sbin/init` for it to be picked up by `systemd-nspawn --boot --directory <path-to-container>`

Reply:
`/boot` should be an implementation detail of certain boot loaders (namely GRUB and UEFI), but the `initScript` boot loader shouldn't require that.